packit/utils/obs_helper.py

```diff
@@ -0,0 +1,291 @@
+        """Creates the package with the name ``package_name`` in the project
+        ``project_name`` on OBS. No sources are uploaded in the process.
+
+        """
+        (root := ET.Element("package")).attrib["name"] = package_name
+        root.attrib["project"] = project_name
+
+        (title := ET.Element("title")).text = f"The {package_name} package"
+        descr = ET.Element("description")
+
+        root.append(title)
+        root.append(descr)
+
+        package_url = core.makeurl(
+            OBSHelper._API_URL,
+            ["source", project_name, package_name, "_meta"],
+        )
+        metafile = core.metafile(package_url, ET.tostring(root))
+        metafile.sync()
+
+    @staticmethod
+    def create_obs_project(
+        project: str,
+        targets: str,
+        owner: Optional[str],
+        package_config: PackageConfig,
+        description: Optional[str],
+    ):
+        conf.get_config()
+        owner = owner or conf.config["api_host_options"][OBSHelper._API_URL]["user"]
+        project_name = project or f"home:{owner}:packit"
+
+        targets_list = targets.split(",")
+        for target in targets_list:
+            if target in DEPRECATED_TARGET_MAP:
+                logger.warning(
+                    f"Target '{target}' is deprecated. "
+                    f"Please use '{DEPRECATED_TARGET_MAP[target]}' instead.",
+                )
+
+        project_metadata = OBSHelper.targets_to_project_meta(
+            targets=targets_list,
+            owner=owner,
+            project_name=project_name,
+            description=description,
+        )
+
+        logger.info(f"Using OBS project name = {project_name}")
+
+        project_url = core.makeurl(
+            OBSHelper._API_URL,
+            ["source", project_name, "_meta"],
+        )
+        metafile = core.metafile(project_url, ET.tostring(project_metadata))
+        metafile.sync()
+
+        package_names = list(package_config.packages.keys())
+
+        if len(package_names) != 1:
+            raise ValueError("Cannot handle multiple packages in package_config")
+
+        OBSHelper.create_package(project_name, (package_name := package_names[0]))
+
+        return project_name, package_name
+
+    @staticmethod
+    def init_project(
+        build_dir: str,
+        package_name: str,
+        project_name: str,  # prj_str
+    ) -> Path:
+        core.Project.init_project(
+            OBSHelper._API_URL,
+            (prj_dir := Path(build_dir)),
+            project_name,
+        )
+
+        (pkg_dir := (prj_dir / package_name)).mkdir()
+        core.checkout_package(
+            OBSHelper._API_URL,
+            project_name,
+            package_name,
+            prj_dir=prj_dir,
+            pathname=pkg_dir,
+        )
+
+        pkg = core.Package(pkg_dir)
+
+        for fname in filter(os.path.isfile, os.listdir(pkg_dir)):
+            pkg.delete_file(fname)
```

Comment:
I found the bug, turns out, I'm an idiot. The correct solution is this:
```suggestion
        for fname in os.listdir(pkg_dir):
            pkg.delete_file(fname)
```

What was happening here? We ran `os.listdir()` on the checked out package directory and I thought I'll be clever and will filter out all directory entries that are not files. The problem is that `os.path.isfile(fname)` (which is run implicitly by `filter`) is **not** run in `pkg_dir` but of course in the local working directory! That appeared to do "the right thing" because I tested this only with packit so far. And packit puts the `packit.spec` and the tarball into the current working directory. So it appeared like the code was working, because I always had the previous tarball and the spec file in the current working directory :facepalm:

Anyway, the above patch fixes it for me:
![image](https://github.com/packit/packit/assets/45594031/3141a0b5-86fc-4344-b28d-2147d18c1c17)